Resizetizer.NT/SkiaSharpSvgTools.cs

```diff
@@ -1,116 +1,50 @@
 ﻿using SkiaSharp;
-using Svg;
 using Svg.Skia;
 using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.Drawing;
-using System.IO;
-using System.Text;
-using System.Text.RegularExpressions;
 
 namespace Resizetizer
 {
-	internal class SkiaSharpSvgTools
+	internal class SkiaSharpSvgTools : SkiaSharpTools, IDisposable
 	{
+		SKSvg svg;
+		SKPaint paint;
+
 		public SkiaSharpSvgTools(SharedImageInfo info, ILogger logger)
+			: base(info, logger)
 		{
-			Info = info;
-			Logger = logger;
-
-			var sw = new Stopwatch();
-			sw.Start();
-
-			var svgDoc = SKSvg.Open(Info.Filename);
-
-			void ChangeFill(SvgElement element)
+			if (Info.TintColor is Color tint)
 			{
-				if (element is SvgPath ePath && ePath.Fill is SvgColourServer eFill)
-				{
-					Logger?.Log($"Found Fill: {eFill.Colour.ToString()}");
-
-					if (!eFill.Colour.IsEmpty)
-					{
-						ePath.Fill = new SvgColourServer(Info.TintColor.Value);
-
-						Logger?.Log($"Changing Fill: {Info.TintColor.ToString()}");
-					}
-				}
+				var color = new SKColor(unchecked((uint)tint.ToArgb()));
+				Logger?.Log($"Detected a tint color of {color}");
 
-				if (element.Children.Count > 0)
+				paint = new SKPaint
 				{
-					foreach (var item in element.Children)
-						ChangeFill(item);
-				}
+					ColorFilter = SKColorFilter.CreateBlendMode(color, SKBlendMode.SrcIn)
```

Comment:
Applying a tint color is basically painting with a color filter of "Source In" to effectively mean, draw the image, but get your color from this instead.

This is better that swapping out the colors in the SVG for a few reasons:
 - less code
 - supports other shapes and things (unless this is not what was wanted)
 - probably faster (but this is an assumption based on SkiaSharp coolness)
 - supports overlapping shapes not changing the tint (if two paths overlap and the color has alpha, then the overlap will be "darker")